components/avatar/avatar.vue

```diff
@@ -203,20 +204,26 @@ export default {
     },
   },
 
-  mounted () {
-    this.init();
+  watch: {
+    defaultSlotContent: {
+      deep: true,
+      immediate: true,
+      handler: function () {
+        this.init();
+      },
+    },
   },
 
   updated () {
-    if (this.kind === 'initials') {
-      this.slottedInitials = this.$slots.default[0].text || this.$slots.default[0].textContent;
-      this.formatInitials(this.slottedInitials);
-    }
+    this.defaultSlotContent = this.$slots.default ? this.$slots.default[0] : null;
```

Comment:
Would replace it with `this.defaultSlotContent = this.$slots.default?.[0] ? this.$slots.default[0] : null;`

Reply:
Mhmmm... `this.defaultSlotContent = this.$slots.default?.[0] ?? null;` this is even better

Reply:
I didn't knew I could do it that way, thanks!